### ISSUE TITLE
start up with loopback dns configured - no route to host

### ISSUE THREAD

**Issue by adamsewell** [0xERR0R/blocky#247]
I'm using a test machine to play around with blocky, which the machine has 127.0.0.1 configured as the DNS server to use blocky. When starting blocky in this scenario, it is unable to download any of the feeds off the get go. I thought that maybe bootstrapDns would be the right setting to enable but it doesn't appear to help. Am I missing something or could blocky use the upstream resolvers to download the feeds on stat up? Thanks!

`[2021-08-22 08:18:43]  INFO list_cache: starting download link=http://sysctl.org/cameleon/hosts
[2021-08-22 08:18:43]  INFO list_cache: starting download link=https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
[2021-08-22 08:18:43]  WARN list_cache: error during file processing: Get "http://sysctl.org/cameleon/hosts": dial tcp: lookup sysctl.org: no such host
[2021-08-22 08:18:43]  WARN list_cache: error during file processing: Get "https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts": dial tcp: lookup raw.githubusercontent.com: no such host
[2021-08-22 08:18:43]  INFO list_cache: group import finished group=ads total_count=0
[2021-08-22 08:18:43]  INFO list_cache: starting download link=https://urlhaus.abuse.ch/downloads/hostfile/
[2021-08-22 08:18:43]  WARN list_cache: error during file processing: Get "https://urlhaus.abuse.ch/downloads/hostfile/": dial tcp: lookup urlhaus.abuse.ch: no such host
[2021-08-22 08:18:43]  INFO list_cache: group import finished group=phishing total_count=0`

**Comment by 0xERR0R** [comment]
Your config looks good. "bootstrapDns" resolver should be used to resolve the addresses from the links. If you add `logLevel: trace` to you config, do you see "using 8.8.4.4:53 as bootstrap dns server"?

**Comment by adamsewell** [comment]
Yes, it's there. 

```
[2021-08-22 20:50:34] DEBUG using 8.8.4.4:53 as bootstrap dns server
[2021-08-22 20:50:34]  INFO list_cache: starting download link=https://urlhaus.abuse.ch/downloads/hostfile/
[2021-08-22 20:50:34]  WARN list_cache: error during file processing: Get "https://urlhaus.abuse.ch/downloads/hostfile/": dial tcp: lookup urlhaus.abuse.ch: no such host
[2021-08-22 20:50:34]  INFO list_cache: group import finished group=phishing total_count=0
[2021-08-22 20:50:34]  INFO list_cache: starting download link=https://block.energized.pro/porn/formats/hosts
[2021-08-22 20:50:34]  WARN list_cache: error during file processing: Get "https://block.energized.pro/porn/formats/hosts": dial tcp: lookup block.energized.pro: no such host
[2021-08-22 20:50:34]  INFO list_cache: group import finished group=porn total_count=0
[2021-08-22 20:50:34]  INFO list_cache: starting download link=https://block.energized.pro/basic/formats/hosts
[2021-08-22 20:50:34]  INFO list_cache: starting download link=https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
[2021-08-22 20:50:34]  WARN list_cache: error during file processing: Get "https://block.energized.pro/basic/formats/hosts": dial tcp: lookup block.energized.pro: no such host
[2021-08-22 20:50:34]  INFO list_cache: starting download link=http://sysctl.org/cameleon/hosts
[2021-08-22 20:50:34]  WARN list_cache: error during file processing: Get "https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts": dial tcp: lookup raw.githubusercontent.com: no such host
[2021-08-22 20:50:34]  WARN list_cache: error during file processing: Get "http://sysctl.org/cameleon/hosts": dial tcp: lookup sysctl.org: no such host
[2021-08-22 20:50:34]  INFO list_cache: group import finished group=ads total_count=0
[2021-08-22 20:50:34]  INFO list_cache: starting download link=https://raw.githubusercontent.com/anudeepND/whitelist/master/domains/whitelist.txt
[2021-08-22 20:50:34]  WARN list_cache: error during file processing: Get "https://raw.githubusercontent.com/anudeepND/whitelist/master/domains/whitelist.txt": dial tcp: lookup raw.githubusercontent.com: no such host
```

**Comment by 0xERR0R** [comment]
interesting, I have only linux machines to test. Is 8.8.4.4 reachable with nslookup (`nslookup raw.githubusercontent.com 8.8.4.4`)

**Comment by 0xERR0R** [comment]
I'll try today on a Windows 10 VM, maybe this is only Windows problem

**Comment by adamsewell** [comment]
That's what I'm testing on as well.

**Comment by 0xERR0R** [comment]
![blocky](https://user-images.githubusercontent.com/25868513/130510732-917f9bf8-d182-4418-9813-a7e5c553b6a7.png)
I tried blocky 0.15 on windows 10 with your config, start was successful.

Maybe windows blocks some connections on your machine?

**Comment by adamsewell** [comment]
Well that is super odd. This is a fresh install of Windows 10 with nothing else installed, no AV, no well... anything installed. And you had set your DNS to manual config with 127.0.0.1 as the DNS server?

**Comment by 0xERR0R** [comment]
No, I just started blocky and tried to perform a request against it with nslookup (nslookup google.com 127.0.0.1) and it worked

**Comment by 0xERR0R** [comment]
I tried again, it worked. After disabling of Ipv6 I could reproduce the behavior. Same error "no such host"

**Comment by adamsewell** [comment]
Well that is super weird now, I just tried it again here at my office and it's working just fine with IPv6 enabled. I don't recall disabling IPv6 but it's possible I did. 

I disabled IPv6 and sure enough, no route to host.

**Comment by 0xERR0R** [comment]
It looks like golang uses under windows always the system resolver. There is an open proposal in golang which addresses this problem: https://github.com/golang/go/issues/33097

As far I understand. it is not possible to use custom DNS resolver in go binaries in windows
